utils/src/server/rpc_module.rs

```diff
@@ -466,6 +466,43 @@ impl<Context: Send + Sync + 'static> RpcModule<Context> {
 		Ok(MethodResourcesBuilder { build: ResourceVec::new(), callback })
 	}
 
+	/// Register a new **blocking** synchronous RPC method, which computes the response with the given callback.
+	/// Unlike the regular [`register_method`](RpcModule::register_method), this method can block its thread and perform expensive computations.
+	pub fn register_blocking_method<R, F>(
+		&mut self,
+		method_name: &'static str,
+		callback: F,
+	) -> Result<MethodResourcesBuilder, Error>
+	where
+		Context: Send + Sync + 'static,
+		R: Serialize,
+		F: Fn(Params, Arc<Context>) -> Result<R, Error> + Copy + Send + Sync + 'static,
+	{
+		let ctx = self.ctx.clone();
+		let callback = self.methods.verify_and_insert(
+			method_name,
+			MethodCallback::new_async(Arc::new(move |id, params, tx, claimed| {
```

Comment:
> dq: Why can't we do `new_sync` i.e, the closure itself it synchronous?
> 
> hrm, I assume using `tokio::spawn_blocking` is more efficient than `std::thread::spawn` because tokio is already initialized when the server is started?

Yeah, tokio will manage a threadpool for blocking tasks, spawning OS threads can be costly (pending on how blocking the callback really is). That said, we can probably go with `new_sync` here anyway, I thought we need to track the future to completion, but we don't really...

> Should we have a way to register an async closure too or is that the footgun you were talking about?

Yeah, that's the footgun. Would have to wrap the async callback into something like `LocalSet` that's bound to the thread, but the moment you do some `.await` on something that might be expecting a regular tokio runtime (for io or timers or something) I'm not sure exactly what would happen, but I wouldn't be surprised to see a panic.

Reply:
Tried to make it work with sync callback, but there are some assumptions about lifetimes for params sync callback makes, plus it also means I need to pass in claimed resources handler, so all in all using async callback is way less invasive for what is ultimately a niche feature.